src/XMakeBuildEngine/Evaluation/LazyItemEvaluator.cs

```diff
@@ -0,0 +1,704 @@
+
+            return ret;
+        }
+
+        public void ProcessItemElement(string rootDirectory, ProjectItemElement itemElement, bool conditionResult)
+        {
+            if (itemElement.Include != null)
+            {
+                ProcessItemElementInclude(rootDirectory, itemElement, conditionResult);
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        void ProcessItemElementInclude(string rootDirectory, ProjectItemElement itemElement, bool conditionResult)
+        {
+            IncludeOperationBuilder operationBuilder = new IncludeOperationBuilder();
+            operationBuilder.ElementOrder = _nextElementOrder++;
+            operationBuilder.ItemElement = itemElement;
+            operationBuilder.RootDirectory = rootDirectory;
+            operationBuilder.ItemType = itemElement.ItemType;
+            operationBuilder.ConditionResult = conditionResult;
+
+            //  Code corresponds to Evaluator.CreateItemsFromInclude
+
+            // STEP 1: Expand properties in Include
+            string evaluatedIncludeEscaped = _outerExpander.ExpandIntoStringLeaveEscaped(itemElement.Include, ExpanderOptions.ExpandProperties, itemElement.IncludeLocation);
+
+            // STEP 2: Split Include on any semicolons, and take each split in turn
+            if (evaluatedIncludeEscaped.Length > 0)
+            {
+                IList<string> includeSplitsEscaped = ExpressionShredder.SplitSemiColonSeparatedList(evaluatedIncludeEscaped);
+
+                foreach (string includeSplitEscaped in includeSplitsEscaped)
+                {
+                    // STEP 3: If expression is "@(x)" copy specified list with its metadata, otherwise just treat as string
+                    bool isItemListExpression;
+                    ProcessSingleItemVectorExpressionForInclude(includeSplitEscaped, operationBuilder, itemElement.IncludeLocation, out isItemListExpression);
+
+                    if (!isItemListExpression)
+                    {
+                        // The expression is not of the form "@(X)". Treat as string
+
+                        //  Code corresponds to EngineFileUtilities.GetFileList
+                        bool containsEscapedWildcards = EscapingUtilities.ContainsEscapedWildcards(includeSplitEscaped);
+                        bool containsRealWildcards = FileMatcher.HasWildcards(includeSplitEscaped);
+
+                        if (containsEscapedWildcards && containsRealWildcards)
+                        {
+                            // Umm, this makes no sense.  The item's Include has both escaped wildcards and 
+                            // real wildcards.  What does he want us to do?  Go to the file system and find
+                            // files that literally have '*' in their filename?  Well, that's not going to 
+                            // happen because '*' is an illegal character to have in a filename.
```

Comment:
I don't get it.  Is it not illegal on non-Windows filesystems?

BTW these comments aren't mine, they're just copied from other code (in this case EngineFileUtilities.GetFileList)